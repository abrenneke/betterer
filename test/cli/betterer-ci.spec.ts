import { ciΔ, startΔ } from '@betterer/cli';
import { unexpectedDiff } from '@betterer/reporter/dist/messages';

import { createFixture } from '../fixture';

const ARGV = ['node', './bin/betterer'];

describe('betterer ci', () => {
  it('should add a diff to the summary if there is any change to the results file', async () => {
    let indexTs = `
const a = 'a';
const one = 1;
`;

    // Test not outputting entire results file (snapshot)
    for (let i = 0; i < 1000; i++) {
      indexTs += 'console.log(a * one);\n';
    }

    const { paths, logs, cleanup, resolve, writeFile } = await createFixture('test-betterer-ci-diff', {
      'src/index.ts': indexTs,
      '.betterer.ts': `
import { typescript } from '@betterer/typescript';

export default {
  'typescript use strict mode': typescript('./tsconfig.json', {
    strict: true
  })
};
      `,
      'tsconfig.json': `
{
  "compilerOptions": {
    "noEmit": true,
    "lib": ["esnext"],
    "moduleResolution": "node",
    "target": "ES5",
    "typeRoots": ["../../node_modules/@types/"],
    "resolveJsonModule": true
  },
  "include": ["./src/**/*", ".betterer.ts"]
}
      `
    });

    const fixturePath = paths.cwd;
    const indexPath = resolve('./src/index.ts');

    await ciΔ(fixturePath, ARGV);

    await writeFile(indexPath, indexTs.split('\n').slice(0, 990).join('\n'));

    const diffSummary = await ciΔ(fixturePath, ARGV);

    expect(diffSummary.expected).not.toBeNull();
    expect(diffSummary.unexpectedDiff).toEqual(true);
    expect(diffSummary.worse).toHaveLength(0);

    expect(logs[logs.length - 1].split('\n').length).toBeLessThan(100);
    expect(logs).toMatchSnapshot();

    await cleanup();
  });

  it('should work with `start` and the CI env variable', async () => {
    const { paths, logs, cleanup, resolve, writeFile } = await createFixture('test-betterer-ci-start', {
      'src/index.ts': `
const a = 'a';
const one = 1;
console.log(a * one);
      `,
      '.betterer.ts': `
import { typescript } from '@betterer/typescript';

export default {
  'typescript use strict mode': typescript('./tsconfig.json', {
    strict: true
  })
};
      `,
      'tsconfig.json': `
{
  "compilerOptions": {
    "noEmit": true,
    "lib": ["esnext"],
    "moduleResolution": "node",
    "target": "ES5",
    "typeRoots": ["../../node_modules/@types/"],
    "resolveJsonModule": true
  },
  "include": ["./src/**/*", ".betterer.ts"]
}
      `
    });

    const fixturePath = paths.cwd;
    const indexPath = resolve('./src/index.ts');

    process.env.CI = '1';

    await startΔ(fixturePath, ARGV);

    await writeFile(indexPath, `const a = 'a';\nconst one = 1;\nconsole.log(one + one);\nconsole.log(a * one);`);

    const diffSummary = await startΔ(fixturePath, ARGV);

    expect(diffSummary.expected).not.toBeNull();
    expect(diffSummary.unexpectedDiff).toEqual(true);
    expect(diffSummary.worse).toHaveLength(0);

    expect(logs).toMatchSnapshot();

    await cleanup();
  });

  it('should work when a test gets worse', async () => {
    const { paths, logs, cleanup, resolve, writeFile } = await createFixture('test-betterer-ci-worse', {
      'src/index.ts': `
const a = 'a';
const one = 1;
console.log(a * one);
      `,
      '.betterer.ts': `
import { typescript } from '@betterer/typescript';

export default {
  'typescript use strict mode': typescript('./tsconfig.json', {
    strict: true
  })
};
      `,
      'tsconfig.json': `
{
  "compilerOptions": {
    "noEmit": true,
    "lib": ["esnext"],
    "moduleResolution": "node",
    "target": "ES5",
    "typeRoots": ["../../node_modules/@types/"],
    "resolveJsonModule": true
  },
  "include": ["./src/**/*", ".betterer.ts"]
}
      `
    });

    const fixturePath = paths.cwd;
    const indexPath = resolve('./src/index.ts');

    await ciΔ(fixturePath, ARGV);

    await writeFile(indexPath, `const a = 'a';\nconst one = 1;\nconsole.log(one * a);\nconsole.log(a * one);`);

    const diffSummary = await ciΔ(fixturePath, ARGV);

    expect(diffSummary.expected).not.toBeNull();
    expect(diffSummary.unexpectedDiff).toEqual(false);
    expect(diffSummary.worse.length).toBeGreaterThan(0);

    expect(logs).toMatchSnapshot();

    await cleanup();
  });

  it('should not output diff if any tests got worse', async () => {
    const { paths, logs, cleanup, resolve, writeFile } = await createFixture('test-betterer-ci-worse-no-diff', {
      'src/index.ts': `
const a = 'a';
const one = 1;
debugger;
console.log(a * one);
      `,
      '.betterer.ts': `
import { typescript } from '@betterer/typescript';
import { eslint } from '@betterer/eslint';

export default {
  'typescript use strict mode': typescript('./tsconfig.json', {
    strict: true
  }),
  'eslint no debugger': eslint({ 'no-debugger': 'error' }).include('./src/**/*.ts')
};
      `,
      'tsconfig.json': `
{
  "compilerOptions": {
    "noEmit": true,
    "lib": ["esnext"],
    "moduleResolution": "node",
    "target": "ES5",
    "typeRoots": ["../../node_modules/@types/"],
    "resolveJsonModule": true
  },
  "include": ["./src/**/*", ".betterer.ts"]
}
      `
    });

    const fixturePath = paths.cwd;

    await ciΔ(fixturePath, ARGV);

    await writeFile(
      resolve('./src/index.ts'),
      `const a = 'a';\nconst one = 1;\ndebugger;\n\nconsole.log(a * one);debugger;`
    );

    const diffSummary = await ciΔ(fixturePath, ARGV);

    expect(diffSummary.unexpectedDiff).toBe(true);
    expect(logs.join('\n')).not.toContain(unexpectedDiff());

    await cleanup();
  });
});
