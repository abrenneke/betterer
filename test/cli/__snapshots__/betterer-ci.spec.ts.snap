// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`betterer ci should add a diff to the summary if there is any change to the results file 1`] = `
Array [
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms):

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
🤔 typescript use strict mode: running \\"typescript use strict mode\\"!

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
✅ typescript use strict mode: \\"typescript use strict mode\\" got checked for the first time! (1,000 issues) 🎉

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ typescript use strict mode: \\"typescript use strict mode\\" got checked for the first time! (1,000 issues) 🎉

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ typescript use strict mode: \\"typescript use strict mode\\" got checked for the first time! (1,000 issues) 🎉

1 test got checked. 🤔
1 test got checked for the first time! 🎉

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms):

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
🤔 typescript use strict mode: running \\"typescript use strict mode\\"!

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
✅ typescript use strict mode: \\"typescript use strict mode\\" got better! (13 fixed issues, 987 remaining) 😍

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ typescript use strict mode: \\"typescript use strict mode\\" got better! (13 fixed issues, 987 remaining) 😍

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ typescript use strict mode: \\"typescript use strict mode\\" got better! (13 fixed issues, 987 remaining) 😍

1 test got checked. 🤔
1 test got better! 😍

Unexpected diff found:
- Expected
+ Result

@@ -1,9 +1,9 @@
  // BETTERER RESULTS V2.
  exports[\`typescript use strict mode\`] = {
    value: \`{
-     \\"src/index.ts:3050135194\\": [
+     \\"src/index.ts:34372536\\": [
        [2, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
        [3, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
        [4, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
        [5, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
        [6, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
@@ -986,23 +986,10 @@
        [983, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
        [984, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
        [985, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
        [986, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
        [987, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [988, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [989, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [990, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [991, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [992, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [993, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [994, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [995, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [996, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [997, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [998, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [999, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [1000, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"],
-       [1001, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"]
+       [988, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"]
      ]
    }\`
  };


",
]
`;

exports[`betterer ci should work when a test gets worse 1`] = `
Array [
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms):

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
🤔 typescript use strict mode: running \\"typescript use strict mode\\"!

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
✅ typescript use strict mode: \\"typescript use strict mode\\" got checked for the first time! (1 issue) 🎉

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ typescript use strict mode: \\"typescript use strict mode\\" got checked for the first time! (1 issue) 🎉

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ typescript use strict mode: \\"typescript use strict mode\\" got checked for the first time! (1 issue) 🎉

1 test got checked. 🤔
1 test got checked for the first time! 🎉

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms):

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
🤔 typescript use strict mode: running \\"typescript use strict mode\\"!

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
・ 1 existing issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\".
🤔 typescript use strict mode: running \\"typescript use strict mode\\"!

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
・ 1 existing issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\".
・ 1 new issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\":
🤔 typescript use strict mode: running \\"typescript use strict mode\\"!

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
・ 1 existing issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\".
・ 1 new issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\":
・
・   <project>/fixtures/test-betterer-ci-worse/src/index.ts
・   2 | const one = 1;
・   3 | console.log(one * a);
・ > 4 | console.log(a * one);
・     |             ^ The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.
・
🤔 typescript use strict mode: running \\"typescript use strict mode\\"!

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
・ 1 existing issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\".
・ 1 new issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\":
・
・   <project>/fixtures/test-betterer-ci-worse/src/index.ts
・   2 | const one = 1;
・   3 | console.log(one * a);
・ > 4 | console.log(a * one);
・     |             ^ The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.
・
🔥 typescript use strict mode: \\"typescript use strict mode\\" got worse. (1 new issue, 2 total) 😔

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
🔥 typescript use strict mode: \\"typescript use strict mode\\" got worse. (1 new issue, 2 total) 😔
・ 1 existing issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\".
・ 1 new issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\":
・
・   <project>/fixtures/test-betterer-ci-worse/src/index.ts
・   2 | const one = 1;
・   3 | console.log(one * a);
・ > 4 | console.log(a * one);
・     |             ^ The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.
・

Error: \\"typescript use strict mode\\" got worse. (1 new issue, 2 total) 😔

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

💥 Betterer (0ms): 1 test done! 1 test errored!
🔥 typescript use strict mode: \\"typescript use strict mode\\" got worse. (1 new issue, 2 total) 😔
・ 1 existing issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\".
・ 1 new issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\":
・
・   <project>/fixtures/test-betterer-ci-worse/src/index.ts
・   2 | const one = 1;
・   3 | console.log(one * a);
・ > 4 | console.log(a * one);
・     |             ^ The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.
・

Error: \\"typescript use strict mode\\" got worse. (1 new issue, 2 total) 😔

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

💥 Betterer (0ms): 1 test done! 1 test errored!
🔥 typescript use strict mode: \\"typescript use strict mode\\" got worse. (1 new issue, 2 total) 😔
・ 1 existing issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\".
・ 1 new issue in \\"<project>/fixtures/test-betterer-ci-worse/src/index.ts\\":
・
・   <project>/fixtures/test-betterer-ci-worse/src/index.ts
・   2 | const one = 1;
・   3 | console.log(one * a);
・ > 4 | console.log(a * one);
・     |             ^ The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.
・

Error: \\"typescript use strict mode\\" got worse. (1 new issue, 2 total) 😔

1 test got checked. 🤔
1 test got worse. 😔


",
]
`;

exports[`betterer ci should work with \`start\` and the CI env variable 1`] = `
Array [
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms):

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
🤔 typescript use strict mode: running \\"typescript use strict mode\\"!

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
✅ typescript use strict mode: \\"typescript use strict mode\\" got checked for the first time! (1 issue) 🎉

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ typescript use strict mode: \\"typescript use strict mode\\" got checked for the first time! (1 issue) 🎉

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ typescript use strict mode: \\"typescript use strict mode\\" got checked for the first time! (1 issue) 🎉

1 test got checked. 🤔
1 test got checked for the first time! 🎉

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms):

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
🤔 typescript use strict mode: running \\"typescript use strict mode\\"!

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
✅ typescript use strict mode: \\"typescript use strict mode\\" stayed the same. (1 issue) 😐

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ typescript use strict mode: \\"typescript use strict mode\\" stayed the same. (1 issue) 😐

",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ typescript use strict mode: \\"typescript use strict mode\\" stayed the same. (1 issue) 😐

1 test got checked. 🤔
1 test stayed the same. 😐

Unexpected diff found:
- Expected
+ Result

  // BETTERER RESULTS V2.
  exports[\`typescript use strict mode\`] = {
    value: \`{
-     \\"src/index.ts:1499252024\\": [
+     \\"src/index.ts:2615232350\\": [
-       [2, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"]
+       [3, 12, 1, \\"The left-hand side of an arithmetic operation must be of type /'any/', /'number/', /'bigint/' or an enum type.\\", \\"177604\\"]
      ]
    }\`
  };


",
]
`;
